linera-storage/src/chain_guards.rs

```diff
@@ -50,10 +51,19 @@ impl ChainGuards {
     /// the same chain.
     pub async fn guard(&self, chain_id: ChainId) -> ChainGuard {
         let guard = self.get_or_create_lock(chain_id);
+        let chain_id_string = chain_id.to_string();
+        let queue_metric = QUEUED_CHAIN_GUARD_REQUESTS.with_label_values(&[&chain_id_string]);
```

Comment:
In theory, we shouldn't use chain IDs as labels because they could be millions of chain ids.

nit: `let labels = [&chain_id.to_string()];` then `with_label_values(&labels)` ?

Reply:
Yeah, it could become a bottleneck pretty quickly. I rewrote the PR to be simpler by measuring just the latency to acquire the lock instead. I guess that matters more for performance, and contention should make this value rise quickly.